static/src/javascripts/projects/commercial/modules/messenger.js

```diff
@@ -0,0 +1,240 @@
+// @flow
+import reportError from 'lib/report-error';
+import dfpOrigin from 'commercial/modules/messenger/dfp-origin';
+import postMessage from 'commercial/modules/messenger/post-message';
+
+const ALLOWED_HOSTS = [dfpOrigin, `${location.protocol}//${location.host}`];
+const LISTENERS = {};
+let REGISTERED_LISTENERS = 0;
+
+const error405 = {
+    code: 405,
+    message: 'Service %% not implemented',
+};
+const error500 = {
+    code: 500,
+    message: 'Internal server error\n\n%%',
+};
+
+type StandardMessage = {
+    id: string,
+    type: string,
+    iframeId: string,
+    value: {
+        height: number,
+        width: number,
+    },
+};
+
+// A legacy from programmatic ads running in friendly iframes. They can
+// on occasion be larger than the size returned by DFP. And so they
+// have been setup to send a message of the form:
+type Payload = {
+    type: string,
+    value: {
+        id: string,
+        height: number,
+        width: number,
+    },
+};
+
+const isProgrammaticMessage = (payload: Payload): boolean =>
+    payload.type === 'set-ad-height' &&
+    'id' in payload.value &&
+    'height' in payload.value;
+
+const toStandardMessage = (payload: Payload): StandardMessage => ({
+    id: 'aaaa0000-bb11-cc22-dd33-eeeeee444444',
+    type: 'resize',
+    iframeId: payload.value.id,
+    value: {
+        height: +payload.value.height,
+        width: +payload.value.width,
+    },
+});
+
+// Incoming messages contain the ID of the iframe into which the
+// source window is embedded.
+const getIframe = (data: Object): HTMLElement | null =>
+    document.getElementById(data.iframeId);
+
+// Until DFP provides a way for us to identify with 100% certainty our
+// in-house creatives, we are left with doing some basic tests
+// such as validating the anatomy of the payload and whitelisting
+// event type
+const isValidPayload = (payload: Object): boolean =>
```

Comment:
I've managed to settle on: 
`const isValidPayload = (payload: StandardMessage): boolean =>`

...

I really like the idea of breaking out some of the responsibilities of `onMessage`, however, maybe this is a job for a subsequent PR. I'd like to get the basic ES6 conversion to PROD first, ensure there are no regressions, then iterate on refactoring the module.

I've refined it to only two incidences of 'Object' type, both associated with the `options` param of the export functions.